File: src/translation/TerminologyDictionary.m
classdef TerminologyDictionary < handle
    % TerminologyDictionary - Manages consistent technical translations
    % between Portuguese and English for scientific articles
    %
    % This class provides a comprehensive mapping of technical terms
    % used in structural engineering, deep learning, and materials science
    % to ensure consistent translation throughout the document.
    
    properties (Access = private)
        structural_engineering_terms
        deep_learning_terms
        materials_science_terms
        statistics_terms
        general_academic_terms
        abbreviations_map
        units_map
    end
    
    methods
        function obj = TerminologyDictionary()
            % Constructor - Initialize all terminology mappings
            obj.initializeStructuralEngineeringTerms();
            obj.initializeDeepLearningTerms();
            obj.initializeMaterialsScienceTerms();
            obj.initializeStatisticsTerms();
            obj.initializeGeneralAcademicTerms();
            obj.initializeAbbreviationsMap();
            obj.initializeUnitsMap();
        end
        
        function english_term = translateTerm(obj, portuguese_term, domain)
            % Translate a Portuguese term to English within specified domain
            %
            % Args:
            %   portuguese_term (string): Portuguese term to translate
            %   domain (string): Domain category ('structural', 'deep_learning', 
            %                   'materials', 'statistics', 'academic', 'abbreviations', 'units')
            %
            % Returns:
            %   english_term (string): English translation or original term if not found
            
            switch lower(domain)
                case 'structural'
                    term_map = obj.structural_engineering_terms;
                case 'deep_learning'
                    term_map = obj.deep_learning_terms;
                case 'materials'
                    term_map = obj.materials_science_terms;
                case 'statistics'
                    term_map = obj.statistics_terms;
                case 'academic'
                    term_map = obj.general_academic_terms;
                case 'abbreviations'
                    term_map = obj.abbreviations_map;
                case 'units'
                    term_map = obj.units_map;
                otherwise
                    error('Unknown domain: %s', domain);
            end
            
            if isKey(term_map, portuguese_term)
                english_term = term_map(portuguese_term);
            else
                english_term = portuguese_term; % Return original if not found
                warning('Term not found in dictionary: %s (domain: %s)', portuguese_term, domain);
            end
        end
        
        function all_terms = getAllTerms(obj, domain)
            % Get all terms for a specific domain
            %
            % Args:
            %   domain (string): Domain category
            %
            % Returns:
            %   all_terms (containers.Map): All terms in the specified domain
            
            switch lower(domain)
                case 'structural'
                    all_terms = obj.structural_engineering_terms;
                case 'deep_learning'
                    all_terms = obj.deep_learning_terms;
                case 'materials'
                    all_terms = obj.materials_science_terms;
                case 'statistics'
                    all_terms = obj.statistics_terms;
                case 'academic'
                    all_terms = obj.general_academic_terms;
                case 'abbreviations'
                    all_terms = obj.abbreviations_map;
                case 'units'
                    all_terms = obj.units_map;
                otherwise
                    error('Unknown domain: %s', domain);
            end
        end
        
        function consistency_report = checkConsistency(obj, text, domain)
            % Check terminology consistency in a text
            %
            % Args:
            %   text (string): Text to check for consistency
            %   domain (string): Primary domain for checking
            %
            % Returns:
            %   consistency_report (struct): Report with inconsistencies found
            
            term_map = obj.getAllTerms(domain);
            portuguese_terms = keys(term_map);
            english_terms = values(term_map);
            
            inconsistencies = {};
            mixed_usage = {};
            
            for i = 1:length(portuguese_terms)
                pt_term = portuguese_terms{i};
                en_term = english_terms{i};
                
                pt_found = contains(text, pt_term, 'IgnoreCase', true);
                en_found = contains(text, en_term, 'IgnoreCase', true);
                
                if pt_found && en_found
                    mixed_usage{end+1} = sprintf('%s / %s', pt_term, en_term);
                elseif pt_found
                    inconsistencies{end+1} = sprintf('Portuguese term found: %s (should be: %s)', pt_term, en_term);
                end
            end
            
            consistency_report.inconsistencies = inconsistencies;
            consistency_report.mixed_usage = mixed_usage;
            consistency_report.is_consistent = isempty(inconsistencies) && isempty(mixed_usage);
        end
    end
    
    methods (Access = private)
        function initializeStructuralEngineeringTerms(obj)
            % Initialize structural engineering terminology mappings
            obj.structural_engineering_terms = containers.Map();
            
            % Basic structural terms
            obj.structural_engineering_terms('vigas W') = 'W-beams';
            obj.structural_engineering_terms('viga W') = 'W-beam';
            obj.structural_engineering_terms('aço ASTM A572 Grau 50') = 'ASTM A572 Grade 50 steel';
            obj.structural_engineering_terms('aço ASTM A572') = 'ASTM A572 steel';
            obj.structural_engineering_terms('corrosão') = 'corrosion';
            obj.structural_engineering_terms('inspeção estrutural') = 'structural inspection';
            obj.structural_engineering_terms('integridade estrutural') = 'structural integrity';
            obj.structural_engineering_terms('deterioração') = 'deterioration';
            obj.structural_engineering_terms('patologia estrutural') = 'structural pathology';
            
            % Corrosion-specific terms
            obj.structural_engineering_terms('corrosão atmosférica') = 'atmospheric corrosion';
            obj.structural_engineering_terms('corrosão uniforme') = 'uniform corrosion';
            obj.structural_engineering_terms('corrosão por pites') = 'pitting corrosion';
            obj.structural_engineering_terms('corrosão galvânica') = 'galvanic corrosion';
            obj.structural_engineering_terms('produtos de corrosão') = 'corrosion products';
            obj.structural_engineering_terms('óxidos de ferro') = 'iron oxides';
            obj.structural_engineering_terms('ferrugem') = 'rust';
            
            % Structural elements
            obj.structural_engineering_terms('elementos estruturais') = 'structural elements';
            obj.structural_engineering_terms('perfis estruturais') = 'structural profiles';
            obj.structural_engineering_terms('soldabilidade') = 'weldability';
            obj.structural_engineering_terms('ductilidade') = 'ductility';
            obj.structural_engineering_terms('resistência mecânica') = 'mechanical strength';
            obj.structural_engineering_terms('tensão de escoamento') = 'yield strength';
            obj.structural_engineering_terms('tensão de ruptura') = 'ultimate tensile strength';
        end
        
        function initializeDeepLearningTerms(obj)
            % Initialize deep learning terminology mappings
            obj.deep_learning_terms = containers.Map();
            
            % Neural network architectures
            obj.deep_learning_terms('redes neurais convolucionais') = 'convolutional neural networks';
            obj.deep_learning_terms('rede neural convolucional') = 'convolutional neural network';
            obj.deep_learning_terms('segmentação semântica') = 'semantic segmentation';
            obj.deep_learning_terms('aprendizado profundo') = 'deep learning';
            obj.deep_learning_terms('mecanismos de atenção') = 'attention mechanisms';
            obj.deep_learning_terms('mecanismo de atenção') = 'attention mechanism';
            obj.deep_learning_terms('gates de atenção') = 'attention gates';
            
            % Training and optimization
            obj.deep_learning_terms('treinamento') = 'training';
            obj.deep_learning_terms('validação cruzada') = 'cross-validation';
            obj.deep_learning_terms('hiperparâmetros') = 'hyperparameters';
            obj.deep_learning_terms('função de perda') = 'loss function';
            obj.deep_learning_terms('otimizador') = 'optimizer';
            obj.deep_learning_terms('taxa de aprendizado') = 'learning rate';
            obj.deep_learning_terms('regularização') = 'regularization';
            obj.deep_learning_terms('dropout') = 'dropout';
            obj.deep_learning_terms('early stopping') = 'early stopping';
            
            % Architecture components
            obj.deep_learning_terms('encoder') = 'encoder';
            obj.deep_learning_terms('decoder') = 'decoder';
            obj.deep_learning_terms('skip connections') = 'skip connections';
            obj.deep_learning_terms('feature maps') = 'feature maps';
            obj.deep_learning_terms('convolução') = 'convolution';
            obj.deep_learning_terms('max pooling') = 'max pooling';
            obj.deep_learning_terms('upsampling') = 'upsampling';
            obj.deep_learning_terms('bottleneck') = 'bottleneck';
            
            % Data processing
            obj.deep_learning_terms('augmentação de dados') = 'data augmentation';
            obj.deep_learning_terms('pré-processamento') = 'preprocessing';
            obj.deep_learning_terms('normalização') = 'normalization';
            obj.deep_learning_terms('dataset') = 'dataset';
            obj.deep_learning_terms('conjunto de dados') = 'dataset';
            obj.deep_learning_terms('anotação manual') = 'manual annotation';
            obj.deep_learning_terms('máscaras binárias') = 'binary masks';
        end
        
        function initializeMaterialsScienceTerms(obj)
            % Initialize materials science terminology mappings
            obj.materials_science_terms = containers.Map();
            
            % Material properties
            obj.materials_science_terms('composição química') = 'chemical composition';
            obj.materials_science_terms('microestrutura') = 'microstructure';
            obj.materials_science_terms('ferrita-perlita') = 'ferrite-pearlite';
            obj.materials_science_terms('laminação controlada') = 'controlled rolling';
            obj.materials_science_terms('grãos refinados') = 'refined grains';
            
            % Corrosion mechanisms
            obj.materials_science_terms('mecanismos de corrosão') = 'corrosion mechanisms';
            obj.materials_science_terms('processo eletroquímico') = 'electrochemical process';
            obj.materials_science_terms('potencial eletroquímico') = 'electrochemical potential';
            obj.materials_science_terms('ambientes agressivos') = 'aggressive environments';
            obj.materials_science_terms('poluentes atmosféricos') = 'atmospheric pollutants';
            
            % Material standards
            obj.materials_science_terms('especificação ASTM') = 'ASTM specification';
            obj.materials_science_terms('norma ASTM') = 'ASTM standard';
            obj.materials_science_terms('propriedades mecânicas') = 'mechanical properties';
            obj.materials_science_terms('relação resistência-peso') = 'strength-to-weight ratio';
        end
        
        function initializeStatisticsTerms(obj)
            % Initialize statistics terminology mappings
            obj.statistics_terms = containers.Map();
            
            % Statistical measures
            obj.statistics_terms('média') = 'mean';
            obj.statistics_terms('desvio padrão') = 'standard deviation';
            obj.statistics_terms('variância') = 'variance';
            obj.statistics_terms('mediana') = 'median';
            obj.statistics_terms('intervalo de confiança') = 'confidence interval';
            obj.statistics_terms('nível de significância') = 'significance level';
            obj.statistics_terms('teste t de Student') = "Student's t-test";
            obj.statistics_terms('teste de hipótese') = 'hypothesis test';
            
            % Evaluation metrics
            obj.statistics_terms('precisão') = 'precision';
            obj.statistics_terms('revocação') = 'recall';
            obj.statistics_terms('acurácia') = 'accuracy';
            obj.statistics_terms('coeficiente Dice') = 'Dice coefficient';
            obj.statistics_terms('índice Jaccard') = 'Jaccard index';
            obj.statistics_terms('verdadeiros positivos') = 'true positives';
            obj.statistics_terms('falsos positivos') = 'false positives';
            obj.statistics_terms('verdadeiros negativos') = 'true negatives';
            obj.statistics_terms('falsos negativos') = 'false negatives';
            
            % Statistical analysis
            obj.statistics_terms('análise estatística') = 'statistical analysis';
            obj.statistics_terms('significância estatística') = 'statistical significance';
            obj.statistics_terms('correlação') = 'correlation';
            obj.statistics_terms('distribuição normal') = 'normal distribution';
            obj.statistics_terms('teste de normalidade') = 'normality test';
        end
        
        function initializeGeneralAcademicTerms(obj)
            % Initialize general academic terminology mappings
            obj.general_academic_terms = containers.Map();
            
            % Research methodology
            obj.general_academic_terms('metodologia') = 'methodology';
            obj.general_academic_terms('objetivo geral') = 'general objective';
            obj.general_academic_terms('objetivos específicos') = 'specific objectives';
            obj.general_academic_terms('revisão da literatura') = 'literature review';
            obj.general_academic_terms('estado da arte') = 'state of the art';
            obj.general_academic_terms('trabalhos relacionados') = 'related work';
            
            % Document structure
            obj.general_academic_terms('resumo') = 'abstract';
            obj.general_academic_terms('palavras-chave') = 'keywords';
            obj.general_academic_terms('introdução') = 'introduction';
            obj.general_academic_terms('conclusões') = 'conclusions';
            obj.general_academic_terms('discussão') = 'discussion';
            obj.general_academic_terms('resultados') = 'results';
            obj.general_academic_terms('referências') = 'references';
            
            % Academic writing
            obj.general_academic_terms('contribuição científica') = 'scientific contribution';
            obj.general_academic_terms('relevância científica') = 'scientific relevance';
            obj.general_academic_terms('impacto prático') = 'practical impact';
            obj.general_academic_terms('limitações do estudo') = 'study limitations';
            obj.general_academic_terms('trabalhos futuros') = 'future work';
            obj.general_academic_terms('implicações práticas') = 'practical implications';
        end
        
        function initializeAbbreviationsMap(obj)
            % Initialize abbreviations and acronyms mappings
            obj.abbreviations_map = containers.Map();
            
            % Technical abbreviations
            obj.abbreviations_map('IoU') = 'Intersection over Union (IoU)';
            obj.abbreviations_map('CNN') = 'Convolutional Neural Network (CNN)';
            obj.abbreviations_map('ReLU') = 'Rectified Linear Unit (ReLU)';
            obj.abbreviations_map('GPU') = 'Graphics Processing Unit (GPU)';
            obj.abbreviations_map('CPU') = 'Central Processing Unit (CPU)';
            obj.abbreviations_map('RAM') = 'Random Access Memory (RAM)';
            
            % Standards and organizations
            obj.abbreviations_map('ASTM') = 'American Society for Testing and Materials (ASTM)';
            obj.abbreviations_map('AISC') = 'American Institute of Steel Construction (AISC)';
            obj.abbreviations_map('ISO') = 'International Organization for Standardization (ISO)';
            
            % File formats and software
            obj.abbreviations_map('JPEG') = 'Joint Photographic Experts Group (JPEG)';
            obj.abbreviations_map('RGB') = 'Red, Green, Blue (RGB)';
            obj.abbreviations_map('TIFF') = 'Tagged Image File Format (TIFF)';
            obj.abbreviations_map('PDF') = 'Portable Document Format (PDF)';
        end
        
        function initializeUnitsMap(obj)
            % Initialize units and measurements mappings
            obj.units_map = containers.Map();
            
            % Mechanical properties units
            obj.units_map('MPa') = 'MPa';
            obj.units_map('ksi') = 'ksi';
            obj.units_map('GPa') = 'GPa';
            obj.units_map('N/mm²') = 'N/mm²';
            
            % Dimensional units
            obj.units_map('mm') = 'mm';
            obj.units_map('cm') = 'cm';
            obj.units_map('m') = 'm';
            obj.units_map('pixels') = 'pixels';
            
            % Temperature and other units
            obj.units_map('°C') = '°C';
            obj.units_map('K') = 'K';
            obj.units_map('%') = '%';
            obj.units_map('anos') = 'years';
            obj.units_map('meses') = 'months';
        end
    end
end

File: src/translation/terminology_dictionary.py
#!/usr/bin/env python3
"""
Terminology Dictionary for Portuguese-English Technical Translation
Provides consistent technical translations for scientific article content.
"""

class TerminologyDictionary:
    """
    Comprehensive dictionary for technical term translations between Portuguese and English.
    Organized by domain for consistent translation across the scientific article.
    """
    
    def __init__(self):
        self.terminology_map = {
            'structural_engineering': {
                # Steel and Materials
                'aço ASTM A572 Grau 50': 'ASTM A572 Grade 50 steel',
                'vigas W': 'W-beams',
                'vigas de aço': 'steel beams',
                'estruturas metálicas': 'metal structures',
                'elementos estruturais': 'structural elements',
                'integridade estrutural': 'structural integrity',
                'segurança estrutural': 'structural safety',
                'inspeção estrutural': 'structural inspection',
                'monitoramento estrutural': 'structural monitoring',
                'patologia estrutural': 'structural pathology',
                
                # Corrosion Terms
                'corrosão': 'corrosion',
                'corrosão atmosférica': 'atmospheric corrosion',
                'corrosão uniforme': 'uniform corrosion',
                'corrosão por pites': 'pitting corrosion',
                'corrosão galvânica': 'galvanic corrosion',
                'produtos de corrosão': 'corrosion products',
                'processos corrosivos': 'corrosive processes',
                'deterioração': 'deterioration',
                'oxidação': 'oxidation',
                'ferrugem': 'rust',
                
                # Inspection and Testing
                'inspeção visual': 'visual inspection',
                'ensaios não destrutivos': 'non-destructive testing',
                'detecção automatizada': 'automated detection',
                'monitoramento': 'monitoring',
                'avaliação': 'assessment',
                'diagnóstico': 'diagnosis',
                
                # Properties and Characteristics
                'propriedades mecânicas': 'mechanical properties',
                'tensão de escoamento': 'yield strength',
                'tensão de ruptura': 'tensile strength',
                'resistência': 'strength',
                'ductilidade': 'ductility',
                'soldabilidade': 'weldability',
                'composição química': 'chemical composition',
                'microestrutura': 'microstructure'
            },
            
            'deep_learning': {
                # Neural Networks
                'redes neurais convolucionais': 'convolutional neural networks',
                'redes neurais profundas': 'deep neural networks',
                'aprendizado profundo': 'deep learning',
                'inteligência artificial': 'artificial intelligence',
                'visão computacional': 'computer vision',
                'processamento de imagens': 'image processing',
                
                # Architectures
                'U-Net': 'U-Net',
                'Attention U-Net': 'Attention U-Net',
                'arquitetura': 'architecture',
                'encoder': 'encoder',
                'decoder': 'decoder',
                'bottleneck': 'bottleneck',
                'skip connections': 'skip connections',
                'attention gates': 'attention gates',
                'mecanismos de atenção': 'attention mechanisms',
                
                # Training and Optimization
                'treinamento': 'training',
                'otimização': 'optimization',
                'otimizador': 'optimizer',
                'função de perda': 'loss function',
                'taxa de aprendizado': 'learning rate',
                'hiperparâmetros': 'hyperparameters',
                'épocas': 'epochs',
                'batch size': 'batch size',
                'validação cruzada': 'cross-validation',
                'early stopping': 'early stopping',
                'overfitting': 'overfitting',
                'regularização': 'regularization',
                'dropout': 'dropout',
                
                # Data Processing
                'pré-processamento': 'preprocessing',
                'augmentação de dados': 'data augmentation',
                'normalização': 'normalization',
                'redimensionamento': 'resizing',
                'dataset': 'dataset',
                'conjunto de dados': 'dataset',
                'anotação manual': 'manual annotation',
                'máscaras binárias': 'binary masks',
                'ground truth': 'ground truth'
            },
            
            'segmentation_metrics': {
                # Evaluation Metrics
                'segmentação semântica': 'semantic segmentation',
                'Intersection over Union': 'Intersection over Union (IoU)',
                'IoU': 'IoU',
                'Coeficiente Dice': 'Dice coefficient',
                'Dice': 'Dice coefficient',
                'precisão': 'precision',
                'revocação': 'recall',
                'recall': 'recall',
                'F1-Score': 'F1-Score',
                'acurácia': 'accuracy',
                'especificidade': 'specificity',
                'sensibilidade': 'sensitivity',
                
                # Statistical Terms
                'verdadeiros positivos': 'true positives',
                'falsos positivos': 'false positives',
                'verdadeiros negativos': 'true negatives',
                'falsos negativos': 'false negatives',
                'matriz de confusão': 'confusion matrix',
                'curva ROC': 'ROC curve',
                'área sob a curva': 'area under the curve',
                'AUC': 'AUC'
            },
            
            'statistics': {
                # Statistical Analysis
                'análise estatística': 'statistical analysis',
                'teste t de Student': "Student's t-test",
                'teste de significância': 'significance test',
                'intervalo de confiança': 'confidence interval',
                'nível de significância': 'significance level',
                'valor p': 'p-value',
                'hipótese nula': 'null hypothesis',
                'hipótese alternativa': 'alternative hypothesis',
                'média': 'mean',
                'desvio padrão': 'standard deviation',
                'mediana': 'median',
                'quartis': 'quartiles',
                'distribuição normal': 'normal distribution',
                'correlação': 'correlation',
                'regressão': 'regression'
            },
            
            'academic_writing': {
                # Document Structure
                'resumo': 'abstract',
                'palavras-chave': 'keywords',
                'introdução': 'introduction',
                'revisão da literatura': 'literature review',
                'metodologia': 'methodology',
                'resultados': 'results',
                'discussão': 'discussion',
                'conclusões': 'conclusions',
                'referências': 'references',
                'bibliografia': 'bibliography',
                
                # Academic Terms
                'objetivo geral': 'general objective',
                'objetivos específicos': 'specific objectives',
                'justificativa': 'rationale',
                'relevância científica': 'scientific relevance',
                'contribuição': 'contribution',
                'limitações': 'limitations',
                'trabalhos futuros': 'future work',
                'estado da arte': 'state of the art',
                'lacuna de conhecimento': 'knowledge gap',
                
                # Research Terms
                'pesquisa': 'research',
                'estudo': 'study',
                'investigação': 'investigation',
                'experimento': 'experiment',
                'protocolo experimental': 'experimental protocol',
                'procedimento': 'procedure',
                'método': 'method',
                'abordagem': 'approach',
                'técnica': 'technique',
                'ferramenta': 'tool'
            },
            
            'figures_tables': {
                # Figure References
                'Figura': 'Figure',
                'Tabela': 'Table',
                'Gráfico': 'Graph',
                'Diagrama': 'Diagram',
                'Fluxograma': 'Flowchart',
                'Esquema': 'Scheme',
                
                # Figure Descriptions
                'arquitetura da rede': 'network architecture',
                'fluxograma da metodologia': 'methodology flowchart',
                'comparação de segmentações': 'segmentation comparison',
                'gráficos de performance': 'performance graphs',
                'curvas de aprendizado': 'learning curves',
                'mapas de atenção': 'attention maps',
                'características do dataset': 'dataset characteristics',
                'configurações de treinamento': 'training configurations',
                'resultados quantitativos': 'quantitative results',
                'análise computacional': 'computational analysis'
            }
        }
        
        # Abbreviations and acronyms
        self.abbreviations = {
            'CNN': 'CNN',
            'GPU': 'GPU',
            'CPU': 'CPU',
            'RAM': 'RAM',
            'RGB': 'RGB',
            'JPEG': 'JPEG',
            'TIFF': 'TIFF',
            'PDF': 'PDF',
            'LaTeX': 'LaTeX',
            'ASTM': 'ASTM',
            'ISO': 'ISO',
            'AISC': 'AISC',
            'MPa': 'MPa',
            'ksi': 'ksi',
            'mm': 'mm',
            'cm': 'cm',
            'kg': 'kg',
            'GPU': 'GPU',
            'CUDA': 'CUDA',
            'cuDNN': 'cuDNN'
        }
        
        # Units and measurements
        self.units = {
            'pixels': 'pixels',
            'megapixels': 'megapixels',
            'bits': 'bits',
            'bytes': 'bytes',
            'MB': 'MB',
            'GB': 'GB',
            'Hz': 'Hz',
            'MHz': 'MHz',
            'GHz': 'GHz',
            'segundos': 'seconds',
            'minutos': 'minutes',
            'horas': 'hours',
            'graus': 'degrees',
            '°C': '°C',
            '%': '%'
        }
    
    def get_translation(self, term, domain=None):
        """
        Get English translation for a Portuguese term.
        
        Args:
            term (str): Portuguese term to translate
            domain (str, optional): Specific domain to search in
            
        Returns:
            str: English translation or original term if not found
        """
        term_lower = term.lower()
        
        if domain and domain in self.terminology_map:
            if term_lower in self.terminology_map[domain]:
                return self.terminology_map[domain][term_lower]
        
        # Search across all domains if not found in specific domain
        for domain_dict in self.terminology_map.values():
            if term_lower in domain_dict:
                return domain_dict[term_lower]
        
        # Check abbreviations
        if term in self.abbreviations:
            return self.abbreviations[term]
        
        # Check units
        if term_lower in self.units:
            return self.units[term_lower]
        
        # Return original term if no translation found
        return term
    
    def get_all_terms_by_domain(self, domain):
        """
        Get all terms for a specific domain.
        
        Args:
            domain (str): Domain name
            
        Returns:
            dict: Dictionary of Portuguese-English term pairs
        """
        return self.terminology_map.get(domain, {})
    
    def add_term(self, portuguese_term, english_term, domain):
        """
        Add a new term to the dictionary.
        
        Args:
            portuguese_term (str): Portuguese term
            english_term (str): English translation
            domain (str): Domain category
        """
        if domain not in self.terminology_map:
            self.terminology_map[domain] = {}
        
        self.terminology_map[domain][portuguese_term.lower()] = english_term
    
    def validate_consistency(self, text):
        """
        Check for terminology consistency in translated text.
        
        Args:
            text (str): Text to validate
            
        Returns:
            list: List of potential consistency issues
        """
        issues = []
        
        # Check for remaining Portuguese terms
        portuguese_indicators = [
            'ção', 'são', 'ões', 'ção', 'mente', 'dade', 'agem',
            'treinamento', 'aprendizado', 'segmentação', 'detecção'
        ]
        
        for indicator in portuguese_indicators:
            if indicator in text.lower():
                issues.append(f"Potential Portuguese text found: '{indicator}'")
        
        return issues